SConstruct

```diff
@@ -251,10 +261,6 @@ if env_base["target"] == "debug":
     # http://scons.org/doc/production/HTML/scons-user/ch06s04.html
     env_base.SetOption("implicit_cache", 1)
 
-if not env_base["tools"]:
-    # Export templates can't run unit test tool.
-    env_base["tests"] = False
-
```

Comment:
I confirm that it works with `tools=no`, thanks for testing this out. 🙂

But I think this is out of scope for this PR. The main idea/concern about this was likely that the tests shouldn't be part of the export templates, so perhaps this should be set to `False` with `release=yes` rather than `tools=no`.

Allowing this to work with `release=yes` is a nice idea though, but requires some more planning.

Reply:
I frankly do not agree that this is out of scope of the the PR as it defeats the purpose of my PR (as I mentioned, to make sure that the program tests correctly, the best way to do so is the testing), but I do agree with disabling `tests` during `target=release` for the reasons you mentioned, so I changed it based on that and made it a `TODO`.

Reply:
Tests are quite experimental at this stage of development and existing tests don't cover most of the code base yet. I think manual, targeted testing can produce better results as of now. I'm not denying the usefulness of tests of course, just saying that those changes are not crucial for this PR.

Reply:
I ended up rethinking about this, and the only times a user would want `tests` is if in the `dev` or `tools` options and not anything else for now due to `tests` being really experimental themselves. I ended up moving it next to disabling tests to keep it explicit in the file itself.